splitio/sync/impression.py

```diff
@@ -95,3 +97,95 @@ def synchronize_counters(self):
         except APIException:
             _LOGGER.error('Exception raised while reporting impression counts')
             _LOGGER.debug('Exception information: ', exc_info=True)
+
+
+class ImpressionSynchronizerAsync(object):
+    """Impressions async synchronizer class."""
+    def __init__(self, impressions_api, storage, bulk_size):
+        """
+        Class constructor.
+
+        :param impressions_api: Impressions Api object to send data to the backend
+        :type impressions_api: splitio.api.impressions.ImpressionsAPI
+        :param storage: Impressions Storage
+        :type storage: splitio.storage.ImpressionsStorage
+        :param bulk_size: How many impressions to send per push.
+        :type bulk_size: int
+
+        """
+        self._api = impressions_api
+        self._impression_storage = storage
+        self._bulk_size = bulk_size
+        self._failed = asyncio.Queue()
+
+    async def _get_failed(self):
+        """Return up to <BULK_SIZE> impressions stored in the failed impressions queue."""
+        imps = []
+        count = 0
+        while count < self._bulk_size and self._failed.qsize() > 0:
+            try:
+                imps.append(await self._failed.get())
```

Comment:
Check if it should be `imps.append(await self._failed.get(False))`

Reply:
The False is related to Threading Queue.Queue class for not blocking
asyncio.Queue does not have it